docs: Updated the quickstart, CLI installation, and CLI upgrading with several fixes from the quickstart user experience sessions (#230)
kurtosis-tech/kurtosis: docs/docs/guides/adding-command-line-completion.md

---
title: Adding Command-Line Completion
sidebar_label: Adding Command-Line Completion
---

<!-- NOTE TO KURTOSIS DEVS: 

This page was created by referencing the kubectl docs:
* https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-bash-linux/
* https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-bash-mac/
* https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-zsh/
* https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-fish/

-->

<!---------- START IMPORTS ------------>

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<!---------- END IMPORTS ------------>

[The Kurtosis CLI](../reference/cli/cli.md) supports command-line completion for `bash`, `zsh`, and `fish`. With completion installed, you will be able to:

- Complete subcommands (e.g. typing `kurtosis` and pressing TAB will suggest subcommands)

> - Complete dynamic arguments (e.g. typing `kurtosis enclave inspect` and pressing TAB will list the names of existing enclaves if any)

Reviewer: ```suggestion
- Complete dynamic arguments (e.g. typing `kurtosis enclave inspect` and pressing TAB will list the names of running enclaves)
```

I believe this is technically *more* correct.

Author: Tab-complete will complete any enclave name, not just running ones (try `enclave stop` then `enclave inspect` tab-complete)
